File: src/dbflute/adf/meta/memberServicedbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type MemberServiceDbm_T struct {
	df.BaseDBMeta
	ColumnMemberServiceId *df.ColumnInfo
	ColumnMemberId *df.ColumnInfo
	ColumnServicePointCount *df.ColumnInfo
	ColumnServiceRankCode *df.ColumnInfo
	ColumnRegisterDatetime *df.ColumnInfo
	ColumnRegisterUser *df.ColumnInfo
	ColumnRegisterProcess *df.ColumnInfo
	ColumnUpdateDatetime *df.ColumnInfo
	ColumnUpdateUser *df.ColumnInfo
	ColumnUpdateProcess *df.ColumnInfo
	ColumnVersionNo *df.ColumnInfo
}

func (b *MemberServiceDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}
func (b *MemberServiceDbm_T) foreignMember() *df.ForeignInfo {
	columns := []*df.ColumnInfo{
		MemberServiceDbm.GetColumnInfoByPropertyName("memberId"),
		MemberDbm.GetColumnInfoByPropertyName("memberId"),
	}

	return b.BaseDBMeta.Cfi("FK_MEMBER_SERVICE_MEMBER", "Member",
		columns, 0, true, false, false, false,
		"", nil, false, "memberServiceAsOne")
}	
func (b *MemberServiceDbm_T) foreignServiceRank() *df.ForeignInfo {
	columns := []*df.ColumnInfo{
		MemberServiceDbm.GetColumnInfoByPropertyName("serviceRankCode"),
		ServiceRankDbm.GetColumnInfoByPropertyName("serviceRankCode"),
	}

	return b.BaseDBMeta.Cfi("FK_MEMBER_SERVICE_SERVICE_RANK_CODE", "ServiceRank",
		columns, 1, false, false, false, false,
		"", nil, false, "memberServiceList")
}	
func (b *MemberServiceDbm_T) CreateForeignInfoMap() {
	b.ForeignInfoMap = make(map[string]*df.ForeignInfo)
	b.ForeignInfoMap["Member"] = b.foreignMember()
	b.ForeignInfoMap["ServiceRank"] = b.foreignServiceRank()
}

func (b *MemberServiceDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var MemberServiceDbm *MemberServiceDbm_T

func Create_MemberServiceDbm() {
	MemberServiceDbm = new(MemberServiceDbm_T)
	MemberServiceDbm.TableDbName = "MEMBER_SERVICE"
	MemberServiceDbm.TableDispName = "MEMBER_SERVICE"
	MemberServiceDbm.TablePropertyName = "memberService"
	MemberServiceDbm.TableSqlName = new(df.TableSqlName)
	MemberServiceDbm.TableSqlName.TableSqlName = "exampledb.dbo.MEMBER_SERVICE"
	MemberServiceDbm.TableSqlName.CorrespondingDbName = MemberServiceDbm.TableDbName
	MemberServiceDbm.Identity=true

	var memberService df.DBMeta
	memberService = MemberServiceDbm
	MemberServiceDbm.DBMeta=&memberService
	memberServiceIdSqlName := new(df.ColumnSqlName)
	memberServiceIdSqlName.ColumnSqlName = "MEMBER_SERVICE_ID"
	memberServiceIdSqlName.IrregularChar = false
	MemberServiceDbm.ColumnMemberServiceId = df.CCI(&memberService, "MEMBER_SERVICE_ID", memberServiceIdSqlName, "", "", "Integer.class", "memberServiceId", "", true, true,true, "int identity", 10, 0, "",false,"","", "","","",false,"int64")
	memberIdSqlName := new(df.ColumnSqlName)
	memberIdSqlName.ColumnSqlName = "MEMBER_ID"
	memberIdSqlName.IrregularChar = false
	MemberServiceDbm.ColumnMemberId = df.CCI(&memberService, "MEMBER_ID", memberIdSqlName, "", "", "Integer.class", "memberId", "", false, false,true, "int", 10, 0, "",false,"","", "member","","",false,"int64")
	servicePointCountSqlName := new(df.ColumnSqlName)
	servicePointCountSqlName.ColumnSqlName = "SERVICE_POINT_COUNT"
	servicePointCountSqlName.IrregularChar = false
	MemberServiceDbm.ColumnServicePointCount = df.CCI(&memberService, "SERVICE_POINT_COUNT", servicePointCountSqlName, "", "", "Integer.class", "servicePointCount", "", false, false,true, "int", 10, 0, "",false,"","", "","","",false,"int64")
	serviceRankCodeSqlName := new(df.ColumnSqlName)
	serviceRankCodeSqlName.ColumnSqlName = "SERVICE_RANK_CODE"
	serviceRankCodeSqlName.IrregularChar = false
	MemberServiceDbm.ColumnServiceRankCode = df.CCI(&memberService, "SERVICE_RANK_CODE", serviceRankCodeSqlName, "", "", "String.class", "serviceRankCode", "", false, false,true, "char", 3, 0, "",false,"","", "serviceRank","","",false,"string")
	registerDatetimeSqlName := new(df.ColumnSqlName)
	registerDatetimeSqlName.ColumnSqlName = "REGISTER_DATETIME"
	registerDatetimeSqlName.IrregularChar = false
	MemberServiceDbm.ColumnRegisterDatetime = df.CCI(&memberService, "REGISTER_DATETIME", registerDatetimeSqlName, "", "", "java.time.LocalDateTime.class", "registerDatetime", "", false, false,true, "datetime", 23, 3, "",false,"","", "","","",false,"df.Timestamp")
	registerUserSqlName := new(df.ColumnSqlName)
	registerUserSqlName.ColumnSqlName = "REGISTER_USER"
	registerUserSqlName.IrregularChar = false
	MemberServiceDbm.ColumnRegisterUser = df.CCI(&memberService, "REGISTER_USER", registerUserSqlName, "", "", "String.class", "registerUser", "", false, false,true, "nvarchar", 200, 0, "",false,"","", "","","",false,"string")
	registerProcessSqlName := new(df.ColumnSqlName)
	registerProcessSqlName.ColumnSqlName = "REGISTER_PROCESS"
	registerProcessSqlName.IrregularChar = false
	MemberServiceDbm.ColumnRegisterProcess = df.CCI(&memberService, "REGISTER_PROCESS", registerProcessSqlName, "", "", "String.class", "registerProcess", "", false, false,true, "nvarchar", 200, 0, "",false,"","", "","","",false,"string")
	updateDatetimeSqlName := new(df.ColumnSqlName)
	updateDatetimeSqlName.ColumnSqlName = "UPDATE_DATETIME"
	updateDatetimeSqlName.IrregularChar = false
	MemberServiceDbm.ColumnUpdateDatetime = df.CCI(&memberService, "UPDATE_DATETIME", updateDatetimeSqlName, "", "", "java.time.LocalDateTime.class", "updateDatetime", "", false, false,true, "datetime", 23, 3, "",false,"","", "","","",false,"df.Timestamp")
	updateUserSqlName := new(df.ColumnSqlName)
	updateUserSqlName.ColumnSqlName = "UPDATE_USER"
	updateUserSqlName.IrregularChar = false
	MemberServiceDbm.ColumnUpdateUser = df.CCI(&memberService, "UPDATE_USER", updateUserSqlName, "", "", "String.class", "updateUser", "", false, false,true, "nvarchar", 200, 0, "",false,"","", "","","",false,"string")
	updateProcessSqlName := new(df.ColumnSqlName)
	updateProcessSqlName.ColumnSqlName = "UPDATE_PROCESS"
	updateProcessSqlName.IrregularChar = false
	MemberServiceDbm.ColumnUpdateProcess = df.CCI(&memberService, "UPDATE_PROCESS", updateProcessSqlName, "", "", "String.class", "updateProcess", "", false, false,true, "nvarchar", 200, 0, "",false,"","", "","","",false,"string")
	versionNoSqlName := new(df.ColumnSqlName)
	versionNoSqlName.ColumnSqlName = "VERSION_NO"
	versionNoSqlName.IrregularChar = false
	MemberServiceDbm.ColumnVersionNo = df.CCI(&memberService, "VERSION_NO", versionNoSqlName, "", "", "Long.class", "versionNo", "", false, false,true, "bigint", 19, 0, "",false,"OptimisticLockType.VERSION_NO","", "","","",false,"int64")

	MemberServiceDbm.ColumnInfoList = new(df.List)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnMemberServiceId)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnMemberId)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnServicePointCount)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnServiceRankCode)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnRegisterDatetime)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnRegisterUser)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnRegisterProcess)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnUpdateDatetime)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnUpdateUser)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnUpdateProcess)
	MemberServiceDbm.ColumnInfoList.Add(MemberServiceDbm.ColumnVersionNo)


	MemberServiceDbm.ColumnInfoMap=make(map[string]int)
	MemberServiceDbm.ColumnInfoMap["memberServiceId"]=0
		MemberServiceDbm.ColumnInfoMap["memberId"]=1
		MemberServiceDbm.ColumnInfoMap["servicePointCount"]=2
		MemberServiceDbm.ColumnInfoMap["serviceRankCode"]=3
		MemberServiceDbm.ColumnInfoMap["registerDatetime"]=4
		MemberServiceDbm.ColumnInfoMap["registerUser"]=5
		MemberServiceDbm.ColumnInfoMap["registerProcess"]=6
		MemberServiceDbm.ColumnInfoMap["updateDatetime"]=7
		MemberServiceDbm.ColumnInfoMap["updateUser"]=8
		MemberServiceDbm.ColumnInfoMap["updateProcess"]=9
		MemberServiceDbm.ColumnInfoMap["versionNo"]=10
	    MemberServiceDbm.PrimaryKey = true
    MemberServiceDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &memberService
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(MemberServiceDbm.ColumnMemberServiceId)

	MemberServiceDbm.PrimaryInfo = new(df.PrimaryInfo)
	MemberServiceDbm.PrimaryInfo.UniqueInfo = ui
	
	MemberServiceDbm.VersionNoFlag = true
	MemberServiceDbm.VersionNoColumnInfo = MemberServiceDbm.ColumnVersionNo
	
	var memberServiceMeta df.DBMeta = MemberServiceDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["MemberService"] = &memberServiceMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/MemberServiceDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.optional.OptionalEntity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of MEMBER_SERVICE. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class MemberServiceDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final MemberServiceDbm _instance = new MemberServiceDbm();
    private MemberServiceDbm() {}
    public static MemberServiceDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((MemberService)et).getMemberServiceId(), (et, vl) -> ((MemberService)et).setMemberServiceId(cti(vl)), "memberServiceId");
        setupEpg(_epgMap, et -> ((MemberService)et).getMemberId(), (et, vl) -> ((MemberService)et).setMemberId(cti(vl)), "memberId");
        setupEpg(_epgMap, et -> ((MemberService)et).getServicePointCount(), (et, vl) -> ((MemberService)et).setServicePointCount(cti(vl)), "servicePointCount");
        setupEpg(_epgMap, et -> ((MemberService)et).getServiceRankCode(), (et, vl) -> ((MemberService)et).setServiceRankCode((String)vl), "serviceRankCode");
        setupEpg(_epgMap, et -> ((MemberService)et).getRegisterDatetime(), (et, vl) -> ((MemberService)et).setRegisterDatetime(ctldt(vl)), "registerDatetime");
        setupEpg(_epgMap, et -> ((MemberService)et).getRegisterUser(), (et, vl) -> ((MemberService)et).setRegisterUser((String)vl), "registerUser");
        setupEpg(_epgMap, et -> ((MemberService)et).getRegisterProcess(), (et, vl) -> ((MemberService)et).setRegisterProcess((String)vl), "registerProcess");
        setupEpg(_epgMap, et -> ((MemberService)et).getUpdateDatetime(), (et, vl) -> ((MemberService)et).setUpdateDatetime(ctldt(vl)), "updateDatetime");
        setupEpg(_epgMap, et -> ((MemberService)et).getUpdateUser(), (et, vl) -> ((MemberService)et).setUpdateUser((String)vl), "updateUser");
        setupEpg(_epgMap, et -> ((MemberService)et).getUpdateProcess(), (et, vl) -> ((MemberService)et).setUpdateProcess((String)vl), "updateProcess");
        setupEpg(_epgMap, et -> ((MemberService)et).getVersionNo(), (et, vl) -> ((MemberService)et).setVersionNo(ctl(vl)), "versionNo");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    protected final Map<String, PropertyGateway> _efpgMap = newHashMap();
    { xsetupEfpg(); }
    @SuppressWarnings("unchecked")
    protected void xsetupEfpg() {
        setupEfpg(_efpgMap, et -> ((MemberService)et).getMember(), (et, vl) -> ((MemberService)et).setMember((OptionalEntity<Member>)vl), "member");
        setupEfpg(_efpgMap, et -> ((MemberService)et).getServiceRank(), (et, vl) -> ((MemberService)et).setServiceRank((OptionalEntity<ServiceRank>)vl), "serviceRank");
    }
    public PropertyGateway findForeignPropertyGateway(String prop)
    { return doFindEfpg(_efpgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "MEMBER_SERVICE";
    protected final String _tableDispName = "MEMBER_SERVICE";
    protected final String _tablePropertyName = "memberService";
    protected final TableSqlName _tableSqlName = new TableSqlName("exampledb.dbo.MEMBER_SERVICE", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnMemberServiceId = cci("MEMBER_SERVICE_ID", "MEMBER_SERVICE_ID", null, null, Integer.class, "memberServiceId", null, true, true, true, "int identity", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnMemberId = cci("MEMBER_ID", "MEMBER_ID", null, null, Integer.class, "memberId", null, false, false, true, "int", 10, 0, null, false, null, null, "member", null, null, false);
//"int64"
    protected final ColumnInfo _columnServicePointCount = cci("SERVICE_POINT_COUNT", "SERVICE_POINT_COUNT", null, null, Integer.class, "servicePointCount", null, false, false, true, "int", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnServiceRankCode = cci("SERVICE_RANK_CODE", "SERVICE_RANK_CODE", null, null, String.class, "serviceRankCode", null, false, false, true, "char", 3, 0, null, false, null, null, "serviceRank", null, null, false);
//"string"
    protected final ColumnInfo _columnRegisterDatetime = cci("REGISTER_DATETIME", "REGISTER_DATETIME", null, null, java.time.LocalDateTime.class, "registerDatetime", null, false, false, true, "datetime", 23, 3, null, false, null, null, null, null, null, false);
//"df.Timestamp"
    protected final ColumnInfo _columnRegisterUser = cci("REGISTER_USER", "REGISTER_USER", null, null, String.class, "registerUser", null, false, false, true, "nvarchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnRegisterProcess = cci("REGISTER_PROCESS", "REGISTER_PROCESS", null, null, String.class, "registerProcess", null, false, false, true, "nvarchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnUpdateDatetime = cci("UPDATE_DATETIME", "UPDATE_DATETIME", null, null, java.time.LocalDateTime.class, "updateDatetime", null, false, false, true, "datetime", 23, 3, null, false, null, null, null, null, null, false);
//"df.Timestamp"
    protected final ColumnInfo _columnUpdateUser = cci("UPDATE_USER", "UPDATE_USER", null, null, String.class, "updateUser", null, false, false, true, "nvarchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnUpdateProcess = cci("UPDATE_PROCESS", "UPDATE_PROCESS", null, null, String.class, "updateProcess", null, false, false, true, "nvarchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnVersionNo = cci("VERSION_NO", "VERSION_NO", null, null, Long.class, "versionNo", null, false, false, true, "bigint", 19, 0, null, false, OptimisticLockType.VERSION_NO, null, null, null, null, false);
//"int64"

    /**
     * MEMBER_SERVICE_ID: {PK, ID, NotNull, int identity(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberServiceId() { return _columnMemberServiceId; }
    /**
     * MEMBER_ID: {UQ, NotNull, int(10), FK to MEMBER}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberId() { return _columnMemberId; }
    /**
     * SERVICE_POINT_COUNT: {NotNull, int(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnServicePointCount() { return _columnServicePointCount; }
    /**
     * SERVICE_RANK_CODE: {NotNull, char(3), FK to SERVICE_RANK}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnServiceRankCode() { return _columnServiceRankCode; }
    /**
     * REGISTER_DATETIME: {NotNull, datetime(23, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegisterDatetime() { return _columnRegisterDatetime; }
    /**
     * REGISTER_USER: {NotNull, nvarchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegisterUser() { return _columnRegisterUser; }
    /**
     * REGISTER_PROCESS: {NotNull, nvarchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegisterProcess() { return _columnRegisterProcess; }
    /**
     * UPDATE_DATETIME: {NotNull, datetime(23, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnUpdateDatetime() { return _columnUpdateDatetime; }
    /**
     * UPDATE_USER: {NotNull, nvarchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnUpdateUser() { return _columnUpdateUser; }
    /**
     * UPDATE_PROCESS: {NotNull, nvarchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnUpdateProcess() { return _columnUpdateProcess; }
    /**
     * VERSION_NO: {NotNull, bigint(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnVersionNo() { return _columnVersionNo; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnMemberServiceId());
        ls.add(columnMemberId());
        ls.add(columnServicePointCount());
        ls.add(columnServiceRankCode());
        ls.add(columnRegisterDatetime());
        ls.add(columnRegisterUser());
        ls.add(columnRegisterProcess());
        ls.add(columnUpdateDatetime());
        ls.add(columnUpdateUser());
        ls.add(columnUpdateProcess());
        ls.add(columnVersionNo());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnMemberServiceId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // -----------------------------------------------------
    //                                        Unique Element
    //                                        --------------
    public UniqueInfo uniqueOf() { return hpcui(columnMemberId()); }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    /**
     * MEMBER by my MEMBER_ID, named 'member'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignMember() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnMemberId(), MemberDbm.getInstance().columnMemberId());
        return cfi("FK_MEMBER_SERVICE_MEMBER", "member", this, MemberDbm.getInstance(), mp, 0, org.dbflute.optional.OptionalEntity.class, true, false, false, false, null, null, false, "memberServiceAsOne", false);
    }
    /**
     * SERVICE_RANK by my SERVICE_RANK_CODE, named 'serviceRank'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignServiceRank() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnServiceRankCode(), ServiceRankDbm.getInstance().columnServiceRankCode());
        return cfi("FK_MEMBER_SERVICE_SERVICE_RANK_CODE", "serviceRank", this, ServiceRankDbm.getInstance(), mp, 1, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "memberServiceList", false);
    }

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============
    public boolean hasIdentity() { return true; }
    public boolean hasVersionNo() { return true; }
    public ColumnInfo getVersionNoColumnInfo() { return _columnVersionNo; }

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.MemberService"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.MemberServiceCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.MemberServiceBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<MemberService> getEntityType() { return MemberService.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public MemberService newEntity() { return new MemberService(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((MemberService)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((MemberService)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
